tests/test_types.py

```diff
@@ -4401,7 +4432,9 @@ def test_custom_generic_containers():
     T = TypeVar('T')
 
     class GenericList(List[T]):
-        pass
+        @classmethod
+        def __get_pydantic_core_schema__(cls, source_type: Any, handler: GetCoreSchemaHandler) -> CoreSchema:
+            return core_schema.no_info_after_validator_function(GenericList, handler(List[get_args(source_type)[0]]))
```

Comment:
These all still work, we just require you to write these two lines of code now to confirm how you want them to work. I imagine we could provide some sort of decorator or mixin that does this for you, something like:

```python
class PydanticCustomSequence:
    @classmethod
    def __get_pydantic_core_schema__(cls, source_type: Any, handler: GetCoreSchemaHandler) -> CoreSchema:
        return core_schema.no_info_after_validator_function(cls, handler(cls[get_args(source_type)[0]]))
```

It would need to be a bit more complex to handle `list` on Python < 3.9, handle bare types (`x: GenericList`), etc.